main.js

```diff
@@ -40,14 +43,12 @@ function createMenu() {
   // The clicks here take a function so that additional parameters such as
   // a pointer to the menu item can be passed.
   let template = [
-    // 2018-09-12 mp: Disabling check for updates until updater talks to omnitruck
-    // so that users don't get message around missing app-update.yml on App startup.
-    //
-    // {
-    //   label: 'Check for updates...',
-    //   click: updater.checkForUpdates
-    // },
-    // {type: 'separator'},
+    {
+      id: 'updateCheck',
+      label: 'Check for updates...',
+      click: () => { requestFromMenu = true;  app.emit('do-update-check') }
```

Comment:
I understand why you have `requestFromMenu` stored as state on the main process. It originally seemed weird to me but now I think it makes sense - the main process is responsible for 'drawing' our tray app and the logic for whether this request came from the menu only lives here. In this sense, `requestFromMenu` should absolutely be state we store in the main process.

I had to look at this for a long time before understanding this was correct. I think I still like the idea that the main process sets the initial application state and then starts a renderer process that is responsible for all the 'drawing' related functions of our tray application. But thats a refactor we can do later.

Reply:
Yep, I like that idea too - for this pass I just wanted to get us switched over.  We can talk about restructuring how we handle render/main process.